### ISSUE TITLE
Call for Testers of v0.7.0

### ISSUE THREAD

**Issue by suzuki-shunsuke** [suzuki-shunsuke/tfaction#1199]
We are planning to release v0.7.0, but we'd like to test it using prerelease versions before releasing it.

https://github.com/suzuki-shunsuke/tfaction/releases/tag/v0.7.0-1

Please read the release note for the detail.

This release includes the significant security improvement, so we'd like to release this as soon as possible.
But at the same time this release includes a small breaking change and a design change about plan file management.
We tested the prerelease version in our test environment, but we'd like to test it in production environments too.

So we really appreciate it if you try the prerelease version in your production environment.
If you have any troubles or feedbacks, please let us know.
Of course we can help you if you have any troubles.
We can collaborate in private channels if necessary.

Please post comments if you can help us!

Thank you!

**Comment by rrreeeyyy** [comment]
Hi, I am @rrreeeyyy from Topotal who reached out to you on Twitter.

We are conducting internal tests for `tfaction 0.7.0-1`.

We have set the permissions of the `GITHUB_TOKEN` to `restricted` at the Organization level.

When trying to execute the apply workflow in `tfaction 0.7.0-1`, we encountered the following error:

```
couldn't fetch workflows for ${org}/${repo}: HTTP 403: Resource not accessible by integration (https://api.github.com/repos/${org}/${repo}/actions/workflows?per_page=100&page=1)
```

I suspect that during the `terraform apply`, when trying to download the plan file, it's executing `gh run list`, but it seems to be failing due to insufficient permissions.

refs: https://github.com/suzuki-shunsuke/tfaction/blob/9789e4b23338f5f262631ae1e0e8f9d0611efd39/terraform-apply/download_plan_file.sh#L12

In our environment, adding the following to the apply workflow made it execute successfully:

```yaml
permissions:
  actions: read
```

Considering others might encounter the same issue, it might be helpful to add the above information to the upgrade guide.

**Comment by suzuki-shunsuke** [comment]
Hi @rrreeeyyy , thank you for your cooperation and great feedback!
We missed the additional permission is needed. Nice catch!
I updated the guide. https://github.com/suzuki-shunsuke/tfaction/releases/tag/v0.7.0-1

```markdown
  - Add the permission `actions: read` to `terraform-apply` action's `github_token` input, which is the GitHub Actions' token `${{github.token}}` by default
    - 📝 The permission is necessary to download plan files from GitHub Actions Artifacts before running `terraform apply`: [ref](https://github.com/suzuki-shunsuke/tfaction/issues/1199#issuecomment-1760735132)
```

<img width="911" alt="image" src="https://github.com/suzuki-shunsuke/tfaction/assets/13323303/1c1eb925-2f0e-414d-8ddc-9b9af3d04f56">

--

terraform-plan action uploads plan files with [actions/upload-artifact](https://github.com/actions/upload-artifact), so maybe it requires some additional permissions. 🤔

https://github.com/suzuki-shunsuke/tfaction/blob/0175054474269fe87609b47e01434f965caddf85/terraform-plan/action.yaml#L45-L50

**Comment by rrreeeyyy** [comment]
> The terraform-plan action uploads plan files with actions/upload-artifact, so maybe it requires some additional permissions. 🤔

Yes, I had the same thought. However, in our environment, the workflow for `terraform plan` runs normally without permissions like `actions: write`. It's possible that by default, one has write permissions to their "own" workflow.

**Comment by suzuki-shunsuke** [comment]
I see. I confirmed `upload-artifacts` worked without any permission. Thank you.
https://github.com/suzuki-shunsuke/test-github-action/actions/runs/6504383011/job/17666241402
https://github.com/suzuki-shunsuke/test-github-action/actions/runs/6504383011/workflow#L7-L14

**Comment by ponkio-o** [comment]
Hi, I found this Issue helpful as I had a same error. Thank you!

BTW, I did a little research on this issue and it seems that the artifact upload API is undocumented 🤔 
https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28

It appears that a different token is used than `GITHUB_TOKEN`
ref. https://github.com/actions/upload-artifact/issues/180

**Comment by kyontan** [comment]
Hi, I'm kyontan, a SRE from StudySapuri / Quipper ( @quipper ) .

We have confirmed that the `suzuki-shunsuke/tfaction@v0.7.0-1` works on our environment w/ no issue so far.

Due for the scale of our deployment, our setup of `apply` workflow is with the token generated from GitHub App, so it's not enough to add `actions: read` in workflow definition, and we needed to add the permission equivalent to `actions: read` to the GitHub App.

Great thanks for your continuous support / improvements on tfaction.

**Comment by suzuki-shunsuke** [comment]
Thank you always, kyontan.
I'm really glad that tfaction works without issues on your environment!

**Comment by suzuki-shunsuke** [comment]
@ponkio-o @rrreeeyyy
Thank you for your cooperation!
We'd like to release v0.7.0 this week if possible, so could you verify v0.7.0-x by this week?
And if you finish the test, please let us know.

We don't want to rush you, so if it's difficult to finish the test by this week, please let us know.
We can postpone the release.

Thank you again!

**Comment by ponkio-o** [comment]
Hi @suzuki-shunsuke 
I'm using v0.7.0-1 in one company's Terraform repository and have not found any problems.
Several developers have submitted PRs and they all seem to be working correctly!

BTW, will this change the policy for IAM Role used in `terraform plan`? For example, the code in [this document](https://suzuki-shunsuke.github.io/tfaction/docs/config/add-working-directory/#aws-iam-role) and [suzuki-shunsuke/terraform-aws-tfaction](https://github.com/suzuki-shunsuke/terraform-aws-tfaction).

**Comment by suzuki-shunsuke** [comment]
Hi @ponkio-o 
thank you for your testing. Sounds good!

> BTW, will this change the policy for IAM Role used in terraform plan? For example, the code in [this document](https://suzuki-shunsuke.github.io/tfaction/docs/config/add-working-directory/#aws-iam-role) and [suzuki-shunsuke/terraform-aws-tfaction](https://github.com/suzuki-shunsuke/terraform-aws-tfaction).

Correct.

- https://github.com/suzuki-shunsuke/terraform-aws-tfaction/pull/48
- https://github.com/suzuki-shunsuke/tfaction-docs/pull/137

**Comment by rrreeeyyy** [comment]
@suzuki-shunsuke 

Thank you for the update.

We've successfully updated to `v0.7.0-1` and conducted tests with `terraform-provider-aws`. Everything in our AWS environment, including plan/apply operations, is functioning as expected.

Regarding tools like `tfmigrate`, we haven't completed our testing. Please let us know if it's essential for these to be tested prior to release; otherwise, we're comfortable with you proceeding with the release as planned.

**Comment by suzuki-shunsuke** [comment]
@rrreeeyyy 

Thank you for your testing!
About tfmigrate, I think it should work well and the test isn't essential.
I tested tfmigrate in my environment and it worked well.

**Comment by suzuki-shunsuke** [comment]
@kyontan @ponkio-o @rrreeeyyy 
Hi all,

we released v0.7.0 just now!

https://github.com/suzuki-shunsuke/tfaction/releases/tag/v0.7.0

Thank you for your help!
